storage/local/storage.go

```diff
@@ -360,11 +371,28 @@ func (s *memorySeriesStorage) GetMetricForFingerprint(fp clientmodel.Fingerprint
 
 // AppendSamples implements Storage.
 func (s *memorySeriesStorage) AppendSamples(samples clientmodel.Samples) {
+	if len(samples) == 0 {
```

Comment:
The check is probably done in 5ns or so, while the codepath below takes at least 200ns or so... GIven there is a mutex, I thought it's worth it in any case.

But the place where it is called has the same check, because it also wants to send to OpenTSDB, and so that's probably the reason why the check is there, to check only once.

Design-wise, I think it's not the caller's duty to muse about expensiveness of a call. It should be handled be the receiver if it's critical. So I changed the caller now, and added a separate check to the opentsdb method.